### PR TITLE
Create products log

### PR DESCRIPTION
**Pull request by guilhermevanzin**
Create products log
 -> Log.cs
 -> Include AddLog in ProductService and SaleService

### CODE REVIEW

File: Sellers.Domain/Models/Log.cs [hunk 1]
```diff
@@ -1,11 +1,47 @@
...
     {
-        public uint Id {get; protected set;}
-        
-        public uint ProductId {get; protected set;}
-        public Product Product {get; protected set;}
-        
+        public uint Id { get; protected set;}
+        public uint ProductId  { get; protected set;}
+        public long Quantity { get; protected set; }
+
```
Comment (JulioDeBastiani) [inline Sellers.Domain/Models/Log.cs]: Include a `CreatedAt` property

File: Sellers.Domain/Models/Log.cs [hunk 2]
```diff
@@ -1,11 +1,47 @@
...
 {
     public class Log
     {
-        public uint Id {get; protected set;}
-        
-        public uint ProductId {get; protected set;}
-        public Product Product {get; protected set;}
-        
+        public uint Id { get; protected set;}
+        public uint ProductId  { get; protected set;}
```
Comment (JulioDeBastiani) [inline Sellers.Domain/Models/Log.cs]: All relations should be navigable

File: Sellers.Domain/Models/Log.cs [hunk 3]
```diff
@@ -1,11 +1,47 @@
...
+        {
+            if (productId == 0)
+                throw new ArgumentOutOfRangeException(nameof(productId));
+
+            ProductId = productId;
+        }
+
+        public void SetQuantity(long quantity)
+        {
+            if (quantity < 0)
```
Comment (JulioDeBastiani) [inline Sellers.Domain/Models/Log.cs]: Use negative quantities to represent diminishing stock

File: Sellers.Services/ProductService.cs [hunk 4]
```diff
@@ -73,9 +77,13 @@ public virtual async Task<ProductViewModel> PutAsync(uint id, ProductUpdateModel
             
             if(updateModel.UpdateQuantity)
                 product.SetQuantity(updateModel.Quantity);
+                
 
             await _productRepository.UpdateAsync(product);
 
+            if(updateModel.UpdateQuantity)
+                await AddLog(id,updateModel.Quantity);
```
Comment (JulioDeBastiani) [inline Sellers.Services/ProductService.cs]: Log the movements and not the current quantity

File: Sellers.Services/SaleService.cs [hunk 5]
```diff
@@ -154,6 +158,9 @@ public virtual async Task<SaleViewModel> UpdateItemAsync(uint id, uint itemId, S
                 item.SetPrice(updateModel.Price.Value);
 
             await _saleRepository.UpdateAsync(sale);
+
+            if(updateModel.Quantity != null)
+                await AddLog(item.ProductId, item.Product.Quantity);
```
Comment (JulioDeBastiani) [inline Sellers.Services/SaleService.cs]: Merge similar ifs

File: Sellers.Domain/Models/Log.cs (hunk 1 above)
Comment (guilhermevanzin) [reply Sellers.Domain/Models/Log.cs]: Fix in 08a59405e1a35f23fc8220f9f272ba291d8680ef

File: Sellers.Domain/Models/Log.cs (hunk 2 above)
Comment (guilhermevanzin) [reply Sellers.Domain/Models/Log.cs]: Fix in 08a59405e1a35f23fc8220f9f272ba291d8680ef

File: Sellers.Domain/Models/Log.cs (hunk 3 above)
Comment (guilhermevanzin) [reply Sellers.Domain/Models/Log.cs]: Fix in 08a59405e1a35f23fc8220f9f272ba291d8680ef

File: Sellers.Services/ProductService.cs (hunk 4 above)
Comment (guilhermevanzin) [reply Sellers.Services/ProductService.cs]: Fix in 08a59405e1a35f23fc8220f9f272ba291d8680ef

File: Sellers.Services/SaleService.cs (hunk 5 above)
Comment (guilhermevanzin) [reply Sellers.Services/SaleService.cs]: Fix in 08a59405e1a35f23fc8220f9f272ba291d8680ef

File: Sellers.Api/Migrations/20200129183107_ChangeDatetoCreatedDate.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        {
+
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+
+        }
+    }
+}
```
Comment (JulioDeBastiani) [inline Sellers.Api/Migrations/20200129183107_ChangeDatetoCreatedDate.cs]: Empty migration

Comment (guilhermevanzin) [reply Sellers.Api/Migrations/20200129183107_ChangeDatetoCreatedDate.cs]: Fix in 14012b8f55f232c190fa8299dd2e877d1f9d98df

**Pull request merged**